src/dispatch/signal/flows.py

```diff
@@ -1,20 +1,26 @@
+from dispatch.auth.models import DispatchUser
+from dispatch.case import flows as case_flows
+from dispatch.case import service as case_service
 from dispatch.case.models import CaseCreate
 from dispatch.database.core import SessionLocal
-from dispatch.project.models import Project
-from dispatch.case import service as case_service
-from dispatch.case import flows as case_flows
 from dispatch.entity import service as entity_service
+from dispatch.project.models import Project
 from dispatch.signal import service as signal_service
 from dispatch.signal.models import SignalInstanceCreate
 
 
-def create_signal_instance(db_session: SessionLocal, project: Project, signal_instance_data: dict):
+def create_signal_instance(
+    db_session: SessionLocal,
+    project: Project,
+    signal_instance_data: dict,
+    current_user: DispatchUser = None,
+):
     """Creates a signal and a case if necessary."""
     signal = signal_service.get_by_variant_or_external_id(
         db_session=db_session,
         project_id=project.id,
-        external_id=signal_instance_data["id"],
-        variant=signal_instance_data.variant,
+        external_id=signal_instance_data.get("id"),
+        variant=signal_instance_data["variant"],
```

Comment:
should it be `.get("variant")` since it's optional field?

Reply:
I think so, technically it's one or the other, I think we may need to eventually re-add something like raw signal back and enforce either 'id' or 'variant' (otherwise we can't do anything with it). But I need to figure out how to add arbitrary keys in a pydantic model (if that's even possible). 

I'm going to merge this for the time being until I can figure that out.